### PR TITLE
fix(chat): Loading a new message page freezes the client

### LINKED ISSUE #21434: Increasing loading time/client freeze when toggling chat panel
**Describe the bug**
When toggling the chat panel, the rendering of chat messages is delayed. For 100+ messages, it takes 2-3 seconds or longer, with time increasing proportionally to the number of messages. In some cases, this results in a complete client freeze for a couple seconds(as shown in the gif below).

**To Reproduce**
1. Join meeting.
2. Open the chat panel.
3. Send at least 50 messages.
4. Toggle the chat panel.
5. Observe the delay in rendering.

**Expected behavior**
The time to render chat messages should remain consistent, regardless of the total number of messages. The existing pagination mechanism should prevent performance degradation.

**Actual behavior**
Rendering time increases with the number of messages, causing delays or even freezing the client.

**Screenshots**
chat-toggle-issue

**BBB version:**
Reproducible on: dev30

### PR DESCRIPTION
**Pull request by JoVictorNunes**
### What does this PR do?

- When the user clicks `Load more messages` the client freezes for a few seconds before rendering the new page. This was happening because the message component was overheaded with hook calls and data access which were triggering an additional render. In order to fix that, all those hook calls within the message component were placed in the parent component (more stable, doesn't rerender frequently) and passed as props to the message component.

### Closes Issue(s)

Closes #21434 

### How to test

1. Start a session.
2. Go to the chat.
3. Type 100+ messages.
4. Scroll up and click 'Load more messages'.

### CODE REVIEW

**Comment by gustavotrott** [comment]
does it close the issue #21434?

**Comment by JoVictorNunes** [comment]
> does it close the issue #21434?

Maybe. I'm not sure yet. Even though they are the same problem, #21434 happens in a different scenario. I'm doing some testing to see if this PR closes it.

**Review by Arthurk12 — CHANGES_REQUESTED** [review]
The performance when toggling the chat panel and loading previous chat pages has significantly improved, reducing loading time. However, I've noticed that when a new message is sent, all messages on the last page are being re-rendered (see gif below). It seems the trigger for this issue might be the callback ref passed to the `ChatMessage` component, which is instantiated each time the component renders.

issue

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/page/component.tsx [hunk 1]
```diff
@@ -202,6 +252,20 @@ const ChatListPage: React.FC<ChatListPageProps> = ({
             ref={(ref) => {
```
[the remark this answers is not in the document]
Comment (JoVictorNunes) [reply bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/page/component.tsx:252]: Nice catch!

**Review by Arthurk12 — APPROVED** [review]

**Pull request merged**